### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by jeluard**

### CODE REVIEW

File: crates/vault-registry/src/lib.rs [hunk 1]
```diff
@@ -245,6 +246,7 @@ pub mod pallet {
         /// * `InsufficientCollateralAvailable` - if the vault does not own enough collateral
         #[pallet::weight(<T as Config>::WeightInfo::withdraw_collateral())]
         #[transactional]
+/// SBPM1 #[transactional] is consider alpha, use carefully. Also it introduces a cost so it might make sense to try to avoid its usage.
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:249]: The alternative is equally complex in terms of time as we need to be very confident that no invalid state transitions are made. We are not opposed to spending time on this and we anyway plan to introduce more formal verification which should highlight such edge-cases. However, I firmly believe `transactional` should be enabled by default to protect naive implementations from exploitation.

File: crates/vault-registry/src/lib.rs [hunk 2]
```diff
@@ -721,6 +723,7 @@ impl<T: Config> Pallet<T> {
         let vault = Vault::new(vault_id.clone(), public_key);
         Self::insert_vault(&vault_id, vault);
 
+        // SBP M1 Impact of this potential failure on previously updated data?
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:726]: Are the remaining comments in this file related to our usage of `transactional`?

File: parachain/src/chain_spec.rs [hunk 3]
```diff
@@ -680,6 +680,7 @@ fn mainnet_genesis(
         vault_registry: VaultRegistryConfig {
             minimum_collateral_vault: vec![(CurrencyId::KSM, 0)],
             punishment_delay: DAYS,
+/// SBP M1 You might want to consider removing explicit mentions of USD
```
Comment (gregdhill) [inline parachain/src/chain_spec.rs:683]: Why?

File: crates/vault-registry/src/lib.rs (hunk 2 above)
Comment (jeluard) [reply crates/vault-registry/src/lib.rs:726]: Right, I didn't verify if all paths leading to the use of those functions were using `transactional`.

**Pull request closed without merging**